### PR TITLE
💫 Basic Hindi tokenization support

### PR DESCRIPTION
**Pull request by ines**
Related issues: #641
Also CC'ing everyone who participated in the above issue:
@kaustubhn, @gauravgr, @SanketSKasar, @souravsingh 

This pull request adds a basic Hindi `Language` class to support tokenization with spaCy. It also includes a getter for the `NORM` attribute that adds the stem word if available (adapted from [here](http://computing.open.ac.uk/Sites/EACLSouthAsia/Papers/p6-Ramanathan.pdf)).

Since Hindi support has been requested a lot in the past, I hope this will make it easier for others to contribute and improve the language data. (I don't know Hindi, so I was only able to add the very basics – feedback and contributions are appreciated!)

## Todos and further ideas

*  Add `LIKE_NUM` and `IS_DIGIT` getters and include the Hindi digit representations (see [here](https://github.com/explosion/spaCy/issues/641#issuecomment-336636719)). They should probably also be included in the `NORM`. One challenge here is to find the best and most performant way to check whether a string consists of Hindi digits.
* Add tokenizer exceptions and [other language data](https://alpha.spacy.io/usage/adding-languages)
* **Add tests!** Even a few simple examples covering text and expected tokens, or text and expected norms (stem words etc.) would be incredibly helpful.

### CODE REVIEW

File: spacy/lang/hi/stop_words.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+कई
+कर
+करता
+करते
+करना
+करने
+करें
+कहते
+कहा
+का
```
Comment (smarthi) [inline spacy/lang/hi/stop_words.py]: Also add 'अर्थात', 'किन्तु', 'परन्तु', 'अपितु', 'कदाचित'  ???  - need a Hindi Pandit to ratify ?

Comment (kaustubhn) [reply spacy/lang/hi/stop_words.py]: Isn't 'कदाचित' Marathi word, I have almost never seen it used in Hindi.

Comment (smarthi) [reply spacy/lang/hi/stop_words.py]: Neh, its in Hindi, Marathi and Sanskrit (check ??).  U definitely see its wide usage in periodic movies and epic hindi serials.

Comment (smarthi) [reply spacy/lang/hi/stop_words.py]: .. but i could be wrong too - i would say let's keep that out of stopwords - if some Hindi literary guy could comment on its frequency of usage and if it deserves to be a StopWord or not - let's go with that

Comment (kaustubhn) [reply spacy/lang/hi/stop_words.py]: Actually the base script used is 'Devanagri' (https://en.wikipedia.org/wiki/Devanagari) so its a valid word in both languages (Hindi, Marathi), Also not sure what qualifies as a valid token in NLP is it just the occurrence or the usage in the actual language? I wish someone with Hindi Literature understanding would take a look that would be helpful.

Comment (smarthi) [reply spacy/lang/hi/stop_words.py]: +!, hence my earlier comment about a Hindi Pandit having to ratify this.

Comment (ines) [reply spacy/lang/hi/stop_words.py]: Stop words don't have to be 100% perfect, but in general (also for tokenizer exceptions and other language data), we always try to make sure spaCy also covers the *actual* usage in the real world. For example, in the English stop words, including the word "thru" is almost more important and relevant than the word "whence" (which is usually pretty uncommon in the type of text people are using with spaCy). So keeping a good balance here is always good 👍

Comment (smarthi) [reply spacy/lang/hi/stop_words.py]: That's a great point @ines, i think we are good then with what we have here. +1 to merge.

Comment (ines) [reply spacy/lang/hi/stop_words.py]: Great – merging this now so it's on `develop` and can be tested! If anyone wants to contribute more data or tests, you can submit a PR to the `develop` branch 😊  [See here](https://alpha.spacy.io/usage/adding-languages) for the usage guide on adding language data to spaCy in v2.0.

**Pull request merged**